tests/Functional/Driver/PDO/MySQL/DeadlockTest.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Doctrine\DBAL\Tests\Functional\Driver\PDO\MySQL;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Driver as DriverInterface;
+use Doctrine\DBAL\Driver\PDO\MySQL\Driver;
+use Doctrine\DBAL\Exception\DeadlockException;
+use Doctrine\DBAL\Tests\Functional\Driver\AbstractDriverTest;
+use Doctrine\DBAL\Tests\TestUtil;
+use ReflectionObject;
+use Throwable;
+
+use function pcntl_fork;
+use function sleep;
+
+class DeadlockTest extends AbstractDriverTest
```

Comment:
Why does the test need to extend `AbstractDriverTest`? It doesn't test the driver.

Reply:
I've made a mistake while reading the docks. Changed it to `FunctionalTestCase`